src/ef_halloween_smoke.c

```diff
@@ -18,17 +18,84 @@ eEC_PROFILE_c iam_ef_halloween_smoke = {
 };
 
 static void eHalloween_Smoke_init(xyz_t pos, int prio, s16 angle, GAME* game, u16 item_name, s16 arg0, s16 arg1) {
-    // TODO
+    eEC_CLIP->make_effect_proc(eEC_EFFECT_HALLOWEEN_SMOKE, pos, NULL, game, &angle, item_name, prio, arg0, arg1);
 }
 
 static void eHalloween_Smoke_ct(eEC_Effect_c* effect, GAME* game, void* ct_arg) {
-    // TODO
+    s16 angle = *(s16*)ct_arg;
+    
+    effect->timer = 18;
+    
+    (effect->scale).x= (effect->scale).y= (effect->scale).z = 0.01f;
+    
+    (effect->velocity).y = 0.25f;
+    (effect->velocity).x = 0.0f * sin_s(angle);
+    (effect->velocity).z = 0.0f * cos_s(angle);
+
+    (effect->acceleration).y = 0.075f;
+    (effect->acceleration).x = (effect->acceleration).z = 0.0f;
 }
 
 static void eHalloween_Smoke_mv(eEC_Effect_c* effect, GAME* game) {
-    // TODO
+    xyz_t_add (&effect->velocity, &effect->acceleration, &effect->velocity);
+    xyz_t_add (&effect->position, &effect->velocity, &effect->position);
 }
 
+extern Gfx ef_haro01_00_modelT[];
+extern u8 ef_dust01_0[];
+extern u8 ef_dust01_1[];
+extern u8 ef_dust01_2[];
+extern u8 ef_dust01_3[];
+
+static u8* eHalloween_texture_table[] = {
+    ef_dust01_0,
+    ef_dust01_1,
+    ef_dust01_2,
+    ef_dust01_3,
+};
+
+// clang-format off
+static u8 eHalloween_2tile_texture_idx[][2] = {
+    {0, 0},
+    {0, 1},
+    {1, 1},
+    {1, 2},
+    {2, 2},
+    {2, 3},
+    {3, 3},
+    {3, 3},
+    {3, 3},
+};
+// clang-format on
+
+// clang-format off
+static u8 eHalloween_prim_f[] = {
+    0,
+    128,
+    255,
+    128,
+    0,
+    128,
+    255,
+    128,
+    0,
+};
+// clang-format on
+
 static void eHalloween_Smoke_dw(eEC_Effect_c* effect, GAME* game) {
-    // TODO
+    s16 frame = 18 - effect->timer;
+    int tex_idx = frame >> 1;
+    int tile_idx0 = eHalloween_2tile_texture_idx[tex_idx][0];
+    int tile_idx1 = eHalloween_2tile_texture_idx[tex_idx][1];
+    
+    OPEN_DISP(game->graph);
+    
+    eEC_CLIP->auto_matrix_xlu_proc(game, &effect->position, &effect->scale);
+    
+    gSPSegment(NEXT_POLY_XLU_DISP, ANIME_1_TXT_SEG, eHalloween_texture_table[tile_idx0]);
```

Comment:
While `ANIME_1_TXT_SEG` is true to be 8, not all the textures that get assigned to the 8 segment belong to the anime1 category

I think this should be changed to `G_MWO_SEGMENT_N` (N being the segment number) instead of using the anime1 define